src/main/java/com/spotify/futures/CompletableFutures.java

```diff
@@ -83,6 +113,27 @@ private CompletableFutures() {
         });
   }
 
+  private static <T> void setupFailFast(
+      FailFast failFast, CompletableFuture<? extends T>[] all, CompletableFuture<Void> allOf) {
+    ExecutionMetadata executionMeta = new ExecutionMetadata(all);
+    for (int i = 0; i < all.length; i++) {
+      all[i].whenComplete(
+          (result, throwable) -> {
+            if (!allOf.isDone()
+                && throwable != null
+                && failFast.failFast(throwable, executionMeta)) {
+              allOf.completeExceptionally(failFast.withThrowable(throwable));
+
+              if (failFast.cancelAll()) {
```

Comment:
This could be moved into the FailFast interface - something like `void FailFast.cancel(all)`

Reply:
This would give access to the array of completable futures to the implementations of the `FailFast` interface (in terms of encapsulation, it would feel rather inaccurate).

This could also lead to unwanted / unexpected behavior (as someone completing normally the remaining incomplete futures in the `cancel` method). 

What do you think?

Reply:
That's true - perhaps it is too easily misused